File: Objective-C/admob/BannerExample/BannerExample/BannerSnippets.m
#import "BannerSnippets.h"

@implementation BannerSnippets

static NSString *const TestAdUnit = @"ca-app-pub-3940256099942544/2435281174";

- (void)createCustomAdSizeWithBannerView:(GADBannerView *)bannerView {
  // [START create_custom_ad_size]
  bannerView.adSize = GADAdSizeFromCGSize(CGSizeMake(250, 250));
  // [END create_custom_ad_size]
}

// [START create_ad_view]
- (void)createAdViewWithAdViewContainer:(UIView *)adViewContainer
                     rootViewController:(UIViewController *)rootViewController {
  GADBannerView *bannerView = [[GADBannerView alloc] initWithAdSize:GADAdSizeBanner];
  bannerView.adUnitID = TestAdUnit;
  bannerView.rootViewController = rootViewController;
  [adViewContainer addSubview:bannerView];
}
// [END create_ad_view]

// [START load_ad]
- (void)loadBannerAdWithBannerView:(GADBannerView *)bannerView {
  // Request a large anchored adaptive banner with a width of 375.
  bannerView.adSize = GADLargeAnchoredAdaptiveBannerAdSizeWithWidth(this, 375);
  [bannerView loadRequest:[GADRequest request]];
}
// [END load_ad]

// [START ad_events]
#pragma mark - GADBannerViewDelegate

- (void)bannerViewDidReceiveAd:(GADBannerView *)bannerView {
  NSLog(@"Banner ad loaded.");
}

- (void)bannerView:(GADBannerView *)bannerView didFailToReceiveAdWithError:(NSError *)error {
  NSLog(@"Banner ad failed to load: %@", error.localizedDescription);
}

- (void)bannerViewDidRecordImpression:(GADBannerView *)bannerView {
  NSLog(@"Banner ad recorded an impression.");
}

- (void)bannerViewDidRecordClick:(GADBannerView *)bannerView {
  NSLog(@"Banner ad recorded a click.");
}

- (void)bannerViewWillPresentScreen:(GADBannerView *)bannerView {
  NSLog(@"Banner ad will present screen.");
}

- (void)bannerViewWillDismissScreen:(GADBannerView *)bannerView {
  NSLog(@"Banner ad will dismiss screen.");
}

- (void)bannerViewDidDismissScreen:(GADBannerView *)bannerView {
  NSLog(@"Banner ad did dismiss screen.");
}
// [END ad_events]

@end

File: Swift/admob/BannerExample/BannerExample/BannerSnippets.swift
import GoogleMobileAds
import UIKit

class BannerSnippets: NSObject, GADBannerViewDelegate {

  let testAdUnit = "ca-app-pub-3940256099942544/2435281174"

  func createCustomAdSize(bannerView: GADBannerView) {
    // [START create_custom_ad_size]
    bannerView.adSize = GADAdSizeFromCGSize(CGSize(width: 250, height: 250))
    // [END create_custom_ad_size]
  }

  // [START create_ad_view]
  func createAdView(adViewContainer: UIView, rootViewController: UIViewController) {
    let bannerView = GADBannerView(adSize: GADAdSizeBanner)
    bannerView.adUnitID = testAdUnit
    bannerView.rootViewController = rootViewController
    adViewContainer.addSubview(bannerView)
  }
  // [END create_ad_view]

  // [START load_ad]
  func loadBannerAd(bannerView: GADBannerView) {
    // Request a large anchored adaptive banner with a width of 375.
    bannerView.adSize = largeAnchoredAdaptiveBanner(width: 375)
    bannerView.load(GADRequest())
  }
  // [END load_ad]

  // [START ad_events]
  // MARK: - GADBannerViewDelegate methods
  func bannerViewDidReceiveAd(_ bannerView: GADBannerView) {
    print("Banner ad loaded.")
  }

  func bannerView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: Error) {
    print("Banner ad failed to load: \(error.localizedDescription)")
  }

  func bannerViewDidRecordImpression(_ bannerView: GADBannerView) {
    print("Banner ad recorded an impression.")
  }

  func bannerViewDidRecordClick(_ bannerView: GADBannerView) {
    print("Banner ad recorded a click.")
  }

  func bannerViewWillPresentScreen(_ bannerView: GADBannerView) {
    print("Banner ad will present screen.")
  }

  func bannerViewWillDismissScreen(_ bannerView: GADBannerView) {
    print("Banner ad will dismiss screen.")
  }

  func bannerViewDidDismissScreen(_ bannerView: GADBannerView) {
    print("Banner ad did dismiss screen.")
  }
  // [END ad_events]
}
